test/infrastructure/docker/exp/internal/docker/nodepool.go

```diff
@@ -183,12 +193,24 @@ func (np *NodePool) machinesMatchingInfrastructureSpec() []*docker.Machine {
 // addMachine will add a new machine to the node pool and update the docker machine pool status.
 func (np *NodePool) addMachine(ctx context.Context) error {
 	instanceName := fmt.Sprintf("worker-%s", util.RandomString(6))
-	externalMachine, err := docker.NewMachine(ctx, np.cluster, instanceName, np.dockerMachinePool.Spec.Template.CustomImage, np.labelFilters)
+	externalMachine, err := docker.NewMachine(ctx, np.cluster, instanceName, np.labelFilters)
 	if err != nil {
 		return errors.Wrapf(err, "failed to create helper for managing the externalMachine named %s", instanceName)
 	}
 
-	if err := externalMachine.Create(ctx, constants.WorkerNodeRoleValue, np.machinePool.Spec.Template.Spec.Version, np.dockerMachinePool.Spec.Template.ExtraMounts); err != nil {
+	// NOTE: FailureDomains don't mean much in CAPD since it's all local, but we are setting a label on
+	// each container, so we can check placement.
+	labels := np.labelFilters
+	if len(np.machinePool.Spec.FailureDomains) > 0 {
+		// For MachinePools placement is expected to be managed by the underlying infrastructure primitive, but
+		// given that there is no such an thing in CAPD, we are picking a random failure domain.
+		randomIndex := rand.Intn(len(np.machinePool.Spec.FailureDomains)) //nolint:gosec
```

Comment:
Out of curiosity, what's the gosec problem with this line?

Reply:
I would guess that the linter doesn't detect that the result of `len()` can't be `0`.

Can we add a comment behind the nolint why we're using nolint here? (i.e. e.g. which case we're ignoring)

Reply:
it is asking about using crypto.rand..

Reply:
Oh nice! :)